### ISSUE TITLE
`useMultiMutation()` mutation on list views

### ISSUE THREAD

**Issue by posva** [posva/pinia-colada#88]
When using a regular `useMutation()` within a list view, you can mutate different entities but you only get access to the loading state of the last one. It would be useful to allow a version that requires a `key` to compute the different properties in a Map and then each value becomes a function instead of a computed: `data(key)`, `isLoading(key)` so we can check the progress

Problems:

- How long should each data be preserved
- Should there be a way to remove the data
- Any difference between errors and successes?

**Comment by tsiotska** [comment]
@posva Is this desired usage example?
```ts
 const { data, isLoading, error, mutate, reset, clearKey } = useMultiMutation({
   mutation: async (id: number) => {
     return await api.deleteItem(id)
   },
    onSuccess: () => {
     console.log('Last mutation succeeded')
   },
   onError: () => {
      console.log('Last mutation failed')
   },
 })
 
 mutate('item-1')
 
  if (isLoading('item-1')) {
  console.log('Loading...')
 }
 
 console.log(data('item-1'))
 console.log(error('item-1'))
 
 reset('item-1')
 clearKey('item-1')
 ```

**Comment by posva** [comment]
Yes, more or less. It should have a required key I think but it's indeed really nice to have it based on parameters when they are serializable. Maybe here the `key` should allow a plain value return instead of forcing an array. What is the purpose of `clearKey()`? I think having `reset()` is enough
